test/schema_test.rb

```diff
@@ -58,6 +58,17 @@ def test_links
       ['list', 'info', 'identify_resource', 'create', 'submit', 'update', 'delete'],
       schema.resource('resource').links.map { |link| link.name })
   end
+
+  def test_duplicate_links
+    duplicate_links_schema = Marshal.load(Marshal.dump(SAMPLE_SCHEMA))
```

Comment:
I know this one looks ugly, but this was the cleanest way I saw to generate a deep copy of a hash in Ruby. Unless we want to pull in `active_support` as a dependency.